src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/resource/CsvDataExtractionController.kt

```diff
@@ -15,14 +15,16 @@ import org.springframework.web.bind.annotation.RestController
 import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBody
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.common.toIsoDate
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.CsvDataExtractionService
+import java.io.OutputStream
 import java.time.LocalDate
+import java.time.temporal.ChronoUnit
 
 @Tag(name = "CSV Data Extraction Controller")
 @RestController
-@RequestMapping(value = ["download-csv"])
+@RequestMapping(value = ["download-csv"], produces = ["text/csv", MediaType.APPLICATION_JSON_VALUE])
```

Comment:
Adding JSON here allows for the controller advice to handle the exception (if thrown).